crates/sui-graphql-rpc/src/server/sui_sdk_data_provider.rs

```diff
@@ -175,6 +181,65 @@ impl DataProvider for SuiClient {
         Ok(connection)
     }
 
+    async fn fetch_checkpoint_connection(
+        &self,
+        first: Option<u64>,
+        after: Option<String>,
+        last: Option<u64>,
+        before: Option<String>,
+    ) -> Result<Connection<String, Checkpoint>> {
+        ensure_forward_pagination(&first, &after, &last, &before)?;
+
+        let count = first.map(|q| q as usize);
+
+        let after_parsed = if let Some(after) = after.map(|x| x.parse::<u64>()) {
+            match after {
+                Ok(after_u64) => Some(SerdeBigInt::from(after_u64)),
+                _ => {
+                    return Err(Error::InvalidCursor(
+                        "Cannot convert after parameter into u64 in the checkpoint connection"
+                            .to_string(),
+                    ))
+                    .extend()
+                }
+            }
+        } else {
+            None
+        };
```

Comment:
[`Option::transpose`](https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose) might help neaten this up.

```suggestion
        let after = after
            .map(|x| SerdeBigInt::from(x.parse::<u64>()?))
            .transpose()
            .map_err(|_| Error::InvalidCursor(
                "Cannot convert after parameter into u64 in the checkpoint connection".to_string(),
            ))?;
```